processor/src/test/resources/fixtures/org/mapstruct/ap/test/bugs/_913/DomainDtoWithNvmsDefaultMapperImpl.java

```diff
@@ -39,18 +39,15 @@ public Domain create(Dto source) {
         Domain domain = new Domain();
 
         if ( source != null ) {
-            Set<Long> set = stringListToLongSet( source.getStringsInitialized() );
```

Comment:
I only removed the null check from the default NCVS after an invocation of a mapping method. I didn't change the logic on other places for when a null check is done or not. That is still being handled in the `handleLocalVarNullCheck`. What I did is just omit the explicit set to a variable when we were not doing a null check. The decision logic is now in Java instead of the tempaltes.

>Somewhere I made a remark that we should distinguish between:

Distinguish where?
Well the names of the wrappers are somehow depicting that:

* source-null-check - For this the `SetterWrapperForCollectionsAndMapsWithNullCheck` is used when a direct assignment is done or the NVCS is `ALWAYS`. It is also done in `ExistingInstanceSetterWrapperForCollectionsAndMaps` which extends `SetterWrapperForCollectionsAndMapsWithNullCheck`. Maybe we need to call it `SetterWrapperForCollectionsAndMapsWithSourceNullCheck`
* target-null-check - Done in the `ExistingInstanceSetterWrapperForCollectionsAndMaps`
* local-var-null-check - Done in `ExistingInstanceSetterWrapperForCollectionsAndMaps` and `SetterWrapperForCollectionsAndMapsWithSourceNullCheck`

From my point of view a local var null check is needed when:
* Presence check with direct assignment - We need a null check before setting, because we use the copy constructor
* Presence check for existing instance mapping - We need the null check because we call `addAll` / `putAll`.
* No Presence check and direct assignment - We use the copy constructor
* No Presence check and NCVS `ALWAYS` - the user requested one

All other cases we don't need one.